### PR TITLE
[RLlib; Docs overhaul] Overhaul of auto-API reference pages (via sphinx autoclass/automodule).

### PR DESCRIPTION
**Pull request by sven1977**
Overhaul of auto-API reference pages (via sphinx autoclass/automodule).
- Create rllib and rllib/package_ref sub-folders inside ray/doc/source
- Re-do (semi-)automatic API reference page generation and retire old reference pages (which were completely automated and thus had lots of flaws in them).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/schedules/linear_schedule.py [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
-    """
-    Linear interpolation between `initial_p` and `final_p`. Simply
-    uses Polynomial with power=1.0.
+    """Linear interpolation between `initial_p` and `final_p`.
+
+    Uses `PolynomialSchedule` with power=1.0.
+
 
-    final_p + (initial_p - final_p) * (1 - `t`/t_max)
+    The forumla is:
```
Comment (gjoliver) [inline rllib/utils/schedules/linear_schedule.py:10]: formula

**Review by gjoliver — APPROVED** [review]
couple of minor questions. looking awesome.
thanks!

File: doc/source/rllib/package_ref/utils/exploration.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Base Exploration class (ray.rllib.utils.exploration.exploration.Exploration)
+-----------------------------------------------------------------------------
+
+.. autoclass:: ray.rllib.utils.exploration.exploration.Exploration
+    :members:
+
+All built-in Exploration classes
+--------------------------------
+
```
[the remark this answers is not in the document]
Comment (sven1977) [reply doc/source/rllib/package_ref/utils/exploration.rst:23]: done

File: rllib/utils/schedules/linear_schedule.py (hunk 1 above)
Comment (sven1977) [reply rllib/utils/schedules/linear_schedule.py:10]: done

**Pull request merged**